lib/Sema/CSSolver.cpp

```diff
@@ -2637,5 +2583,45 @@ bool ConstraintSystem::solveSimplified(
   auto tooComplex = getExpressionTooComplex(solutions) &&
     !getASTContext().isSwiftVersion3();
 
-  return tooComplex || !firstSolvedConstraint;
+  return tooComplex || !lastSolvedChoice;
+}
+
+bool DisjunctionChoice::solve(SmallVectorImpl<Solution> &solutions,
+                              FreeTypeVariableBinding allowFreeTypeVariables) {
+  CS->simplifyDisjunctionChoice(Choice);
+  return !CS->solveRec(solutions, allowFreeTypeVariables);
+}
+
+DisjunctionChoice::DisjunctionChoice(ConstraintSystem *cs,
+                                     Constraint *constraint)
+    : CS(cs), Choice(constraint) {
+  if (constraint->getKind() != ConstraintKind::BindOverload)
+    return;
+
+  auto choice = constraint->getOverloadChoice();
+  if (choice.getKind() != OverloadChoiceKind::Decl)
+    return;
+
+  auto decl = choice.getDecl();
+  auto &ctx = decl->getASTContext();
+
+  IsUnavailable = decl->getAttrs().isUnavailable(ctx);
+  if (!decl->isOperator())
+    return;
+
+  auto interfaceType = decl->getInterfaceType();
+  IsGenericOperator = interfaceType->is<GenericFunctionType>();
+
+  auto func = dyn_cast<FuncDecl>(constraint->getOverloadChoice().getDecl());
+  auto paramList =
+      func->getParameterList(func->getDeclContext()->isTypeContext());
+  if (paramList->size() != 2) {
+    IsSymmetricOperator = true;
+  } else {
+    auto firstType =
+        paramList->get(0)->getInterfaceType()->getLValueOrInOutObjectType();
+    auto secondType =
+        paramList->get(1)->getInterfaceType()->getLValueOrInOutObjectType();
+    IsSymmetricOperator = firstType->isEqual(secondType);
+  }
```

Comment:
I don't think this makes a lot of sense to do. The entire body of the constructor exists to support hacks to speed things up which we would like to have go away. I'd rather see the logic in the accessors and comments saying these should go away in time.